pkg/v1/tkg/clusterclient/clusterclient.go

```diff
@@ -542,12 +542,21 @@ func (c *client) WaitForClusterInitialized(clusterName, namespace string) error
 		err = currentClusterInfo.RetrievalError
 
 		if err == nil {
-			// If cluster's ReadyCondition is False and severity is Error, it implies non-retriable error, so return error
+			// should retry in this condition
+			// For dimension env, sometimes it will take some time to connect to vcenter to create resources like cluster scret
 			if conditions.IsFalse(currentClusterInfo.ClusterObject, capi.ReadyCondition) &&
 				(*conditions.GetSeverity(currentClusterInfo.ClusterObject, capi.ReadyCondition) == capi.ConditionSeverityError) {
-				return true, errors.Errorf("cluster creation failed, reason:'%s', message:'%s'",
-					conditions.GetReason(currentClusterInfo.ClusterObject, capi.ReadyCondition),
-					conditions.GetMessage(currentClusterInfo.ClusterObject, capi.ReadyCondition))
+				reason := conditions.GetReason(currentClusterInfo.ClusterObject, capi.ReadyCondition)
+				message := conditions.GetMessage(currentClusterInfo.ClusterObject, capi.ReadyCondition)
+
+				maxTimeoutCounter++
+				// if exceeds maxTimeout, return error
+				if interval*time.Duration(maxTimeoutCounter) > maxTimeout {
```

Comment:
two comments here:
1. so the "side-effect" is: for a real "error" level failure, it takes more time to fail ? is it possible to add this logic for vSphere if it is not yet
2. how should we determine a suitable maxTimeout value ?

Reply:
1. correct, it will retry several times before failure. Initially I added retry for `VCenterUnreachable` only. But then I found out fuji capz pipeline failed due to `reason: Failed @ Machine/github-5516-mc-control-plane-xtzgp`. Thus I changed to retry for all kinds of `reason`. For now seems like the machine failure is kind of like pipeline issue. But I'm not sure if other conditions should also be retried. I think it's better to retry than failed by some other unknown `reason`. :) But sure, if it's better to retry only when `VCenterUnreachable`, I could change it back.
2. I re-use the timeout retry counter for `WaitForClusterInitialized`. Hopefully it might be a suitable value since it's been used for a long while.

Reply:
Thanks for the explanations.

As a dot release to fix must have issues, maybe just fix specific issue that has already been found, and we are confidential for the impact ? In this case, maybe just limited to vsphere unless we found similar issues on the other platforms? (do we want/need to expand tests to public cloud?) 

For the main branch, which has more test cycles, seems good to keep current design to see the impact.

Feel free to evaluate baed on your observations.

Reply:
Jeff, do you mean we need to find the real root-cause of why "vCenterUnreachable" happened? instead of just adding retry? thanks.

Reply:
@renmaosheng : retry seems good for an urgent patch. 
`vCenterUnreachable` happened for a slow "internal" env inside lab, and we want to support that. Instead of adding some logic to wait for VC to be ready before reconciling, @hongtaiwang 's fix here is lightweight, and seems good for an urgent fix, but maybe better to limit to vSphere's scenario ONLY, given test team may just verify vSphere platform for this improvement.